lib/hanami/assets/helpers.rb

```diff
@@ -125,9 +153,23 @@ def self.included(base)
       #   <%= javascript 'application' %>
       #
       #   # <script src="https://assets.bookshelf.org/assets/application-28a6b886de2372ee3922fcaf3f78f2d8.js" type="text/javascript"></script>
-      def javascript(*sources)
+      def javascript(*sources, **options)
         _safe_tags(*sources) do |source|
-          html.script(src: _typed_asset_path(source, JAVASCRIPT_EXT), type: JAVASCRIPT_MIME_TYPE).to_s
+          tag_options = options.merge({
+            src: _typed_asset_path(source, JAVASCRIPT_EXT),
+            type: JAVASCRIPT_MIME_TYPE
+          })
+
+          if _sri? || tag_options.include?(:integrity)
+            tag_options[:integrity] ||= _sri_value(source, JAVASCRIPT_EXT)
+            tag_options[:crossorigin] ||= CROSSORIGIN_ANONYMOUS
+          end
+
+          html.script(
+            src: tag_options.delete(:src),
```

Comment:
I've done this to maintain the order of attributes, and not break the existing tests.